src/gui/torrentcontentmodel.cpp

```diff
@@ -497,33 +530,25 @@ void TorrentContentModel::setupModelData(const BitTorrent::AbstractFileStorage &
     qDebug("Torrent contains %d files", filesCount);
     m_filesIndex.reserve(filesCount);
 
-    TorrentContentModelFolder *currentParent;
+    FolderAdaptor::FolderAdaptorPtr rootFolderAdaptor = std::make_shared<FolderAdaptor>(m_rootItem);
```

Comment:
```suggestion
    auto rootFolderAdaptor = std::make_shared<FolderAdaptor>(m_rootItem);
```
The right side provides enough info about the type.

Reply:
I wouldn't even use `FolderAdaptor::FolderAdaptorPtr` alias outside the `FolderAdaptor` declaration. It has no meaning (it doesn't even save the space).

Reply:
rename it to Ptr? it will become `FolderAdaptor::Ptr`.

Reply:
I don't think hiding the fact that it is std::shared_ptr is a good idea.